packages/outputarea/src/widget.ts

```diff
@@ -301,6 +297,20 @@ export class OutputArea extends Widget {
 
     let layout = this.layout as PanelLayout;
     layout.addWidget(panel);
+
+    /**
+     * Wait for the stdin to complete, add it to the model (so it persists)
+     * and remove the stdin widget.
+     */
+    input.value.then(value => {
+      // Use stdin as the stream so it does not get combined with stdout.
+      this.model.add({
+        output_type: 'stream',
+        name: 'stdin',
+        text: value
```

Comment:
I think that `text: value` could use a linebreak ('value + `\n`'). Otherwise the resulting stream output can run together:
![image](https://user-images.githubusercontent.com/5728311/42523256-18d31410-8422-11e8-88c7-99fd2557e11f.png)